### PR TITLE
ENH update to cfitsio 3.470

### LINKED ISSUE #260: upgrade to cfitsio 3.470
It appears that cfitsio has released another version. We should upgrade.

### PR DESCRIPTION
**Pull request by at88mph**
Added cfitsio3470 and altered the setup.py script to use it.

Closes #260

### CODE REVIEW

**Comment by beckermr** [comment]
This should replace the older version.

**Comment by beckermr** [comment]
Also, we ship some patches to the library, some of which may have been pushed upstream. You will need to check those as well.

**Comment by at88mph** [comment]
OK, changing this to a WIP for today.  Thanks.

File: cfitsio3470/docs/changes.txt [hunk 1]
```diff
@@ -1,4 +1,93 @@
...
+  
+  - histo.c uses double precision internally for all floating point
+    binning; new double-precision subroutines fits_calc_binningd(),
+    fits_rebin_wcsd(), and fits_make_histd(); existing
+    single-precision histogram functions still work but convert values
+    to double-precision internally.
+
+  - new subroutine fits_copy_cols() / ffccls() to copy multiple columns
+  
+  - Fix in imcompress.c for HCOMPRESS and PLIO compression of unsigned
```
Comment (beckermr) [inline cfitsio3470/docs/changes.txt]: There is a patch for this bug in particular that may need to be removed.

Comment (at88mph) [reply cfitsio3470/docs/changes.txt]: OK.  Does the Travis build fail if the patches fail?

Comment (beckermr) [reply cfitsio3470/docs/changes.txt]: Unclear. You will need to go through the patches by hand and check the source to make sure they still work. These patches are the least-worst solution.

**Comment by at88mph** [comment]
Thanks for the feedback.  Do you want to manage the version in `setup.py` yourself?

**Comment by beckermr** [comment]
I'm not sure what you mean. The cfitsio version or the fitsio version?

**Comment by at88mph** [comment]
The `fitsio` version.  I would assume it would move to `1.0.6`.

**Comment by beckermr** [comment]
ahhhhhh - we should hold off on that. There may be other PRs.

**Comment by at88mph** [comment]
Thank you.  Does the `cfitsio` source contain the patched version or the pristine one?

**Comment by beckermr** [comment]
The source folder in the repo should be pristine. It is then patched on the fly on install.

File: patches/putcols.c.patch [hunk 2]
```diff
@@ -1,19 +0,0 @@
...
-+             if ( fits_use_standard_strings() ) {
-+                 if (snull[0] == ASCII_NULL_UNDEFINED) {
-+                     *buffer = ' ';
-+                 } else {
-+                     *buffer = '\0';
-+                 }
-+             } else {
-+                 *buffer = ' ';
-+             }
-+
```
Comment (beckermr) [inline patches/putcols.c.patch]: This patch is needed for string handling in python 3. We cannot remove it.

Comment (beckermr) [reply patches/putcols.c.patch]: unless the upstream file has been patched, which as far as I know it has not been.

Comment (at88mph) [reply patches/putcols.c.patch]: It doesn't apply to 3470.  Should it be adapted to it then?

Comment (at88mph) [reply patches/putcols.c.patch]: The line number was off by one.  Fixed now and re-added.

**Comment by beckermr** [comment]
I double-checked the patches by hand and they all appear to apply properly to the new source code.

I think we are good to merge this. @esheldon do you have any thoughts?

**Comment by esheldon** [comment]
I'm fine with it

**Pull request merged**